src/Group.h

```diff
@@ -13,6 +15,31 @@ struct Hub;
 struct UpgradeInfo {
     char *path, *subprotocol;
     size_t pathLength, subprotocolLength;
+    uWS::HTTPParser httpParser;
+
+    std::map<std::string, std::string> getRequestHeaders() {
+        char* curCursor = httpParser.cursor;
+        httpParser.cursor = httpParser.headerBegin;
+        std::map<std::string, std::string> headers;
+
+        for (httpParser++; httpParser.key.second; httpParser++) {
+            std::string key;
+            key.resize(httpParser.key.second);
+
+            size_t i = 0;
+            for (; i < httpParser.key.second; i++) {
+                key[i] = tolower(httpParser.key.first[i]);
+            }
+            key[i] = 0;
```

Comment:
This stores 0 out of bounds (and it is not needed).

You can use std::transform(key.begin(), key.end(), key.begin(), ::tolower); instead since you already have the key as a std::string.

So you could do like this:
std::string key(httpParser.key.first, httpParser.key.second);
std::transform(key.begin(), key.end(), key.begin(), ::tolower);
headers[key] = ...